content/hardware/06.nicla/boards/nicla-voice/tutorials/ei-intruder-detector/content.md

```diff
@@ -0,0 +1,474 @@
+BLEUnsignedCharCharacteristic batteryLevelChar("2A19",                // standard 16-bit characteristic UUID
+                                               BLERead | BLENotify);  // remote clients will be able to get notifications if this characteristic changes
+```
+The Nicla Voice's integrated NDP needs some files stored in the external flash of the board which are:
+- mcu_fw_120_v91.synpkg
+- dsp_firmware_v91.synpkg
+- ei_model.synpkg
+
+> The files must be named exactly as showed below.
+To store them in the Nicla Voice memory you must use the [Syntiant Uploader](assets/Syntiant_Uploader.zip), these files won't be overwritten after any sketch update.
+
+
+```arduino
+  // Neural Decision Processor firmware and ML model files loading
+  NDP.begin("mcu_fw_120_v91.synpkg");
+  NDP.load("dsp_firmware_v91.synpkg");
+  NDP.load("ei_model.synpkg");
+  Serial.println("packages loaded");
+  NDP.getInfo();
+  Serial.println("Configure mic");
+  NDP.turnOnMicrophone();
+```
+The main responsibility of the Nicla Voice code is to listen to and identify the trained sounds, the below code section is in charge of comparing the inferred category and taking a certain action between them. In the case the Nicla detects the door opening, the label parameter will turn to "NN0:opened", this will trigger a BLE alert sending command followed by a listening pause to avoid duplicated alerts and a green LED blinking to visually indicate the event. The same for the "NN0:forcing" label, with the difference that in this case the LED flashes red and the alert message changes.
+
+```arduino
+/**
+  Inference Interruption Callback to be executed with every triggered inference,
+  it controls the built-in LED's and send the alerts through BLE.
+  
+  Possible labels: NN0:opened, NN0:forcing
+
+  Alerts: 1 = mild alert (for door opened), 2 = high alert (for intruder detected)
+
+  @param label The inferred category label
+*/
+void BLEsend(char* label) {
+
+  if (strcmp(label, "NN0:opened") == 0) {
+    alertLevel.writeValue(1);
+    NDP.noInterrupts();
+    nicla::leds.begin();
+    nicla::leds.setColor(green);
+    delay(3000);
+    nicla::leds.end();
+    NDP.interrupts();
+  }
+  if (strcmp(label, "NN0:forcing") == 0) {
+    alertLevel.writeValue(2);
+    NDP.noInterrupts();
+    nicla::leds.begin();
+    nicla::leds.setColor(red);
+    delay(3000);
+    nicla::leds.end();
+    NDP.interrupts();
+  }
+  if (!lowestPower) {
+    Serial.println(label);
+  }
+}
+```
+These methods are used to update the characteristics of this application.
+
+```arduino
+alertLevel.writeValue(x);  // x = 1 if the door is opened or x = 2 if the door is forced
+batteryLevelChar.writeValue(HigherBatteryLevel);
+```
+
+### Portenta H7 Code
+
+The Portenta H7 needs different libraries to work, in this case, are:
+
+- `ArduinoIoTCloud.h` This one handles the IoT Cloud connection and project variables publishing, it can be installed directly from the Arduino Library Manager.
```

Comment:
```suggestion
- `ArduinoIoTCloud.h` This one handles the Arduino Cloud connection and project variables publishing. It can be installed directly from the Arduino Library Manager.
```